pkg/build/builder/docker.go

```diff
@@ -68,9 +68,17 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, initialURLCheckTimeout, os.Stdin, d.gitClient)
 	if err != nil {
-		d.build.Status.Phase = api.BuildPhaseFailed
-		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
-		d.build.Status.Message = api.StatusMessageFetchSourceFailed
+		switch {
+		case strings.Contains(err.Error(), "no such file or directory"):
```

Comment:
can't you just check if it's an os.IsNotExist error like you did in fetchSource?